pkg/network/usm/istio.go

```diff
@@ -0,0 +1,221 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package usm
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
+	"github.com/DataDog/datadog-agent/pkg/process/monitor"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	manager "github.com/DataDog/ebpf-manager"
+)
+
+// envoyCmd represents the search term used for determining
+// whether or not a given PID represents an Envoy process.
+// The search is done over the /proc/<pid>/cmdline file.
```

Comment:
why cmdline? why not the full path?

Reply:
`cmdline` does contain the full path in the context of Istio containers.
This was taken from a Istio container:
```
/usr/local/bin/envoy -cetc/istio/proxy/envoy-rev.json --drain-time-s45 --drain-strategyimmediate --local-address-ip-versionv4 --file-flush-interval-msec1000 --disable-hot-restart --allow-unknown-static-fields --log-for
```
On top of that I think we can also leverage some arguments such as `-cetc/istio/proxy/envoy-rev.json` to reduce the number of false positives (although this will have to be left for a future PR)